nixos/modules/system/boot/loader/raspberrypi/raspberrypi.nix

```diff
@@ -16,28 +16,80 @@ let
 
   platform = pkgs.stdenv.platform;
 
+  builder_uboot = import ./builder_uboot.nix { inherit config; inherit pkgs; };
+
+  builder =
+    if cfg.uboot.enable then
+      "${builder_uboot} -g ${toString cfg.uboot.configurationLimit} -t ${timeoutStr} -c"
+    else
+      builder_generic;
+
+  blCfg = config.boot.loader;
+  timeoutStr = if blCfg.timeout == null then "-1" else toString blCfg.timeout;
+
 in
 
 {
   options = {
 
-    boot.loader.raspberryPi.enable = mkOption {
-      default = false;
-      type = types.bool;
-      description = ''
-        Whether to create files with the system generations in
-        <literal>/boot</literal>.
-        <literal>/boot/old</literal> will hold files from old generations.
-      '';
-    };
+    boot.loader.raspberryPi = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Whether to create files with the system generations in
+          <literal>/boot</literal>.
+          <literal>/boot/old</literal> will hold files from old generations.
+        '';
+      };
 
-    boot.loader.raspberryPi.version = mkOption {
-      default = 2;
-      type = types.enum [ 1 2 3 ];
-      description = ''
-      '';
-    };
+      version = mkOption {
+        default = 2;
+        type = types.enum [ 1 2 3 ];
+        description = ''
+        '';
+      };
 
+      uboot = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable using uboot as bootmanager for the raspberry pi.
+          '';
+        };
+
+        configurationLimit = mkOption {
+          default = 20;
+          example = 10;
+          type = types.int;
+          description = ''
+            Maximum number of configurations in the boot menu.
+          '';
+        };
+
+        startx = mkOption {
```

Comment:
I think a generic solution to set any possible configuration option would be better instead of just these two. E.g. either make it an option set like:
````
boot.loader.raspberryPi.firmwareConfig = { start_x = 1; arm_freq = 950; }
````
or just as plain text:
````
boot.loader.raspberryPi.firmwareConfig = ''
  arm_freq=950
'';
````